### PR TITLE
Add support for Redis Clusters 

### LINKED ISSUE #1696: [CON-99] [Enhancement] Support Redis Cluster
Support Redis Cluster for increased high availability

<sub>[CON-99](https://linear.app/convoy-engineering/issue/CON-99/[enhancement]-support-redis-cluster)</sub>

### PR DESCRIPTION
**Pull request by jirevwe**
Fixes #1696

### CODE REVIEW

File: internal/pkg/rdb/rdb.go [hunk 1]
```diff
@@ -2,40 +2,54 @@ package rdb
...
+
+		client = redis.NewClient(opts)
+		// Add Instrumentation
+		client.AddHook(nrredis.NewHook(opts))
+	} else {
+		client = redis.NewUniversalClient(&redis.UniversalOptions{
+			Addrs: addresses,
+		})
+		// Add Instrumentation
+		client.AddHook(nrredis.NewHook(nil))
```
Comment (subomi) [inline internal/pkg/rdb/rdb.go:45]: @jirevwe Will this retain instrumentation for the cluster?

**Review by subomi — APPROVED** [review]

File: internal/pkg/rdb/rdb.go (hunk 1 above)
Comment (jirevwe) [reply internal/pkg/rdb/rdb.go:45]: It should, though it'll have to be tested to confirm it. The param is the options which is the redis dsn in parts we don't need that for a redis cluster as the API doesn't return it

Comment (jirevwe) [reply internal/pkg/rdb/rdb.go:45]: Test it, it's retained

Comment (subomi) [reply internal/pkg/rdb/rdb.go:45]: Make sense!

**Review by danvixent — APPROVED** [review]

**Pull request merged**